### PR TITLE
#6529: fix starter brick duplication on save; normalize mod definition structure on save

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Fixes one likely cause of starter brick duplication on save
  - Save code was using `isEqual` which is too strict of a check because it returns false if any structure normalization has been performed (e.g., normalizing the starter mod availability declaration)
  - Adds `deleteUnusedStarterBrickDefinitions` to save helpers as a stop-gap 
- Makes normalized availability declaration structure stricter (all properties are now required)
- Modifies save round trip tests to compare normalized versions of the mods
- Fixed bug in `checkAvailability` where the url argument wasn't being passed to checks

## Remaining Work

- [x] Fix local variable naming conventions in save helpers file
- [x] Modify save helpers to exclude unused stater brick definitions
- [x] Add Jest test to confirm fixed behavior

## Checklist

- [x] Add jest or playwright tests and/or storybook stories
- [x] Designate a primary reviewer: @grahamlangford

### CODE REVIEW

File: src/bricks/available.ts [hunk 1]
```diff
@@ -79,27 +94,25 @@ function testSelector(selector: string): boolean {
...
-  const urlPatterns = rawUrlPatterns ? castArray(rawUrlPatterns) : [];
-  const selectors = rawSelectors ? castArray(rawSelectors) : [];
+  const { matchPatterns, urlPatterns, selectors } =
+    normalizeAvailability(availability);
 
   if (process.env.DEBUG) {
     const result = {
       matchPatterns:
-        matchPatterns.length === 0 || testMatchPatterns(matchPatterns),
+        matchPatterns.length === 0 || testMatchPatterns(matchPatterns, url),
```
Comment (twschiller) [inline src/bricks/available.ts:112]: 🐛  bug fix for urls not being passed through

File: src/bricks/available.ts [hunk 2]
```diff
@@ -20,7 +20,22 @@ import { castArray } from "lodash";
...
+
+  return {
+    matchPatterns: castArray(matchPatterns),
+    urlPatterns: castArray(urlPatterns),
+    selectors: castArray(selectors),
+  };
+}
 
 export function testMatchPatterns(
   patterns: string[],
```
Comment (grahamlangford) [inline src/bricks/available.ts:41]: NIT: remove the default to `document.location.href` inside the function and change the type of url to `string | undefined` to enforce the second parameter

File: src/bricks/available.ts (hunk 1 above)
Comment (grahamlangford) [reply src/bricks/available.ts:112]: See comment above to help prevent this in the future

File: src/pageEditor/hooks/useCheckModStarterBrickInvariants.ts [hunk 3]
```diff
@@ -97,7 +98,10 @@ function useCheckModStarterBrickInvariants(): (
           Object.values(cleanModComponent.definitions).some(
             (definitionFromComponent) =>
               !definitionsFromMod.some((definitionFromMod) =>
-                isEqual(definitionFromComponent, definitionFromMod),
+                isInnerDefinitionEqual(
```
Comment (grahamlangford) [inline src/pageEditor/hooks/useCheckModStarterBrickInvariants.ts:101]: NIT: add test for scenario where `isEqual` returns false but `isInnerDefinitionEqual` returns true

**Review by grahamlangford — APPROVED** [review]
Left some suggestions, but I'm good to merge as-is

File: src/pageEditor/hooks/useCheckModStarterBrickInvariants.ts (hunk 3 above)
Comment (twschiller) [reply src/pageEditor/hooks/useCheckModStarterBrickInvariants.ts:101]: IIRC, it's covered now by virtue of use normalizing the shape

**Pull request merged**